### PR TITLE
refactor(protocol-designer): annotate types of `connect`'d components

### PR DESCRIPTION
**Pull request by mcous**
## overview

Companion PR to #5825 but for react-redux `connect`'d components

## changelog

- refactor(protocol-designer): annotate types of `connect`'d components

## review requests

- Make sure changes are types-only

## risk assessment

Low if changes are types-only

### CODE REVIEW

File: protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js [hunk 1]
```diff
@@ -38,8 +38,8 @@ type OP = {| fieldName: TipOffsetFields, className?: string |}
 
 type SP = {|
   disabled: boolean,
-  mmFromBottom: ?string,
-  wellDepthMm: ?number,
+  mmFromBottom: number,
```
Comment (mcous) [inline protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:41]: This ones are a little scary: there's a difference between the types that `mapStateToProps` exports here and the prop types that `TipPositionInput` expects and is coded for. Definitely looks like there's a `NaN` risk in there

Comment (IanLondon) [reply protocol-designer/src/components/StepEditForm/fields/TipPositionField/index.js:41]: Good catch, tracked in new ticket #5830

**Review by IanLondon — APPROVED** [review]
:notebook_with_decorative_cover:

**Review by shlokamin — APPROVED** [review]
LGTM

**Pull request merged**